src/clr-angular/forms/combobox/option-items.directive.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { NgForOf, NgForOfContext } from '@angular/common';
+import {
+  Directive,
+  DoCheck,
+  Input,
+  IterableDiffer,
+  IterableDiffers,
+  TemplateRef,
+  TrackByFunction,
+  ViewContainerRef,
+  OnDestroy,
+} from '@angular/core';
+
+import { OptionSelectionService } from './providers/option-selection.service';
+import { Subscription } from 'rxjs';
+import { ClrPopoverPositionService } from '../../utils/popover/providers/popover-position.service';
+
+@Directive({
+  selector: '[clrOptionItems][clrOptionItemsOf]',
+})
+export class ClrOptionItems<T> implements DoCheck, OnDestroy {
+  private iterableProxy: NgForOf<T>;
+  private _rawItems: T[];
+  private filteredItems: T[];
+  private subscriptions: Subscription[] = [];
+  private filter = '';
+  private _filterField: string;
+  private differ: IterableDiffer<T> | null = null;
+
+  @Input('clrOptionItemsOf')
+  public set rawItems(items: T[]) {
+    this._rawItems = items ? items : [];
+    this.updateItems();
+  }
+
+  @Input('clrOptionItemsTrackBy')
+  set trackBy(value: TrackByFunction<T>) {
+    this.iterableProxy.ngForTrackBy = value;
+  }
+
+  @Input('clrOptionItemsField')
+  public set field(field: string) {
+    this._filterField = field;
+    this.optionService.displayField = field;
+  }
+
+  constructor(
+    public template: TemplateRef<NgForOfContext<T>>,
+    private differs: IterableDiffers,
+    private optionService: OptionSelectionService<T>,
+    private positionService: ClrPopoverPositionService,
+    private vcr: ViewContainerRef
+  ) {
+    this.iterableProxy = new NgForOf<T>(this.vcr, this.template, this.differs);
+    this.subscriptions.push(
+      optionService.inputChanged.subscribe(filter => {
```

Comment:
It needs to test that htis subscription is properly fired and the side effects are correct. E/g this.filter is properly updated when the subscription streams a new event and that the updateItems method is fired.

Reply:
The updateItems method and this.filter property are private. I disagree that we should be testing private code that is implementation specific, as this most often results in inflexible, non-portable tests that break once we decide to change/improve the implementation. 
Let's talk about this.

Reply:
We agreed on a call that this specific case is covered by functional scenarios and the calls inside the observable handler are private and implementation specific.